ui/test/nuxeo-document-comment-thread.test.js

```diff
@@ -0,0 +1,389 @@
+import { pressAndReleaseKeyOn, tap } from '@polymer/iron-test-helpers/mock-interactions';
+
+/* eslint-disable no-unused-expressions */
+suite('<nuxeo-document-comment-thread>', () => {
+  let element;
+
+  setup(async () => {
+    // Arrange
+    element = await fixture(
+      html`
+        <nuxeo-document-comment-thread uid="doc-id"></nuxeo-document-comment-thread>
+      `,
+    );
+    await flush();
+  });
+
+  suite('Visibility', () => {
+    suite('Input Container', () => {
+      test('Should display input area when thread level is less than 3', () => {
+        expect(isElementVisible(element.root.querySelector('#inputContainer'))).to.be.true;
+      });
+
+      test('Should not display input area when thread level is greater than 2', async () => {
+        element.set('level', 3);
+        await flush();
+
+        expect(isElementVisible(element.root.querySelector('#inputContainer'))).to.be.false;
+      });
+
+      test('Should not display edition actions when the input is empty', () => {
+        const inputArea = dom(element.root).querySelector('.input-area');
+        expect(isElementVisible(inputArea.querySelector('[name="submit"]'))).to.be.false;
+        expect(isElementVisible(inputArea.querySelector('[name="clear"]'))).to.be.false;
+      });
+
+      test('Should not display edition actions when the input is blank', async () => {
+        const inputArea = dom(element.root).querySelector('.input-area');
+        inputArea.querySelector('#inputContainer').value = '                ';
+        await flush();
+
+        expect(isElementVisible(inputArea.querySelector('[name="submit"]'))).to.be.false;
+        expect(isElementVisible(inputArea.querySelector('[name="clear"]'))).to.be.false;
+      });
+
+      test('Should display edition actions when the input has non-blank content', async () => {
+        const inputArea = dom(element.root).querySelector('.input-area');
+        inputArea.querySelector('#inputContainer').value = 'This is my new comment';
+        await flush();
+
+        expect(isElementVisible(inputArea.querySelector('[name="submit"]'))).to.be.true;
+        expect(isElementVisible(inputArea.querySelector('[name="clear"]'))).to.be.true;
+      });
+    });
+
+    suite('Listing Comments', () => {
+      test('Should not display any comment when thread has an empty array of comments', () => {
+        expect(dom(element.root).querySelectorAll('nuxeo-document-comment').length).to.equal(0);
+      });
+
+      test('Should display all the comments when less than 10 comments are returned from server', async () => {
+        const comment = {
+          'entity-type': 'comment',
+          parentId: 'doc-id',
+          id: 'comment-id',
+          numberOfReplies: 0,
+          author: 'John Doe',
+          creationDate: '2019-12-09',
+          modificationDate: undefined,
+          text: 'This is my testing comment',
+        };
+
+        sinon.stub(element.$.commentRequest, 'get').resolves({
```

Comment:
overall I agree that we should either stub methods of the element being tested or at connection level (client or http). stubbing things in an internal ´nuxeo-resource` is too tight coupling... if we could use import maps and use stub elements for testing it would be interesting though

Reply:
Stubbing element's internal/private methods is tricky since we can easily defeat the purpose of the test... We can explore that possibility but I'd be more in favor of going to a more abstract thing like http.